fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -188,10 +187,55 @@ public HiveTableStats getTableStats() throws DdlException {
         return result;
     }
 
-    public void refreshTableCache() throws DdlException {
-        Catalog.getCurrentCatalog().getHiveRepository()
-                .refreshTableCache(resourceName, hiveDb, hiveTable, getPartitionColumns(),
-                        new ArrayList<>(nameToColumn.keySet()));
+    public void refreshTableCache(String dbName, String tableName) throws DdlException {
+        org.apache.hadoop.hive.metastore.api.Table table = Catalog.getCurrentCatalog().getHiveRepository()
+                .getTable(resourceName, this.hiveDb, this.hiveTable);
+        List<FieldSchema> unPartHiveColumns = table.getSd().getCols();
+        List<FieldSchema> partHiveColumns = table.getPartitionKeys();
+        Map<String, FieldSchema> allHiveColumns = unPartHiveColumns.stream()
+                .collect(Collectors.toMap(FieldSchema::getName, fieldSchema -> fieldSchema));
+        for (FieldSchema hiveColumn : partHiveColumns) {
+            allHiveColumns.put(hiveColumn.getName(), hiveColumn);
+        }
+        boolean needRefreshColumn = allHiveColumns.size() != this.fullSchema.size();
```

Comment:
I think there are some tricks to judging whether a table has schema changes by the number of columns. If a user adds a column and drop a column, Both are equal. So I think you may judge if each column is the same.